app/assets/javascripts/discourse/app/components/sidebar/api-section.js

```diff
@@ -1,10 +1,47 @@
 import Component from "@glimmer/component";
 import { getOwner, setOwner } from "@ember/application";
+import { inject as service } from "@ember/service";
 
 export default class SidebarApiSection extends Component {
+  @service sidebarState;
+
   constructor() {
     super(...arguments);
     this.section = new this.args.sectionConfig();
     setOwner(this.section, getOwner(this));
   }
+
+  get shouldDisplay() {
+    if (!this.sidebarState.currentPanel.filterable) {
+      return true;
+    }
+    const shouldDisplay =
+      this.sidebarState.filter.length === 0 || this.filteredLinks.length > 0;
+    const index = this.sidebarState.filteredOutSections.indexOf(
+      this.section.name
+    );
+    if (shouldDisplay) {
+      if (index !== -1) {
+        this.sidebarState.filteredOutSections.removeObject(this.section.name);
+      }
+    } else {
+      if (index === -1) {
+        this.sidebarState.filteredOutSections.pushObject(this.section.name);
+      }
+    }
+    return shouldDisplay;
+  }
+
+  get filteredLinks() {
+    if (this.sidebarState.filter) {
+      return this.section.links.filter((link) => {
+        return link.text
```

Comment:
It could be nice to also match on `this.section.title` as well; then for example all of the Email section links like rejected, received would show when you type Email

Reply:
Also if we wanted to have keywords on links (we will...), this would be the place to search them

Reply:
Sounds good, if title match, then I will display all links

Reply:
Yah fair enough :+1: